__tests__/distributors/local.test.ts

```diff
@@ -0,0 +1,173 @@
+    utilsExtractJdkFile.mockImplementation(() => 'some/random/path/');
+
+    // Spy on path methods
+    resolvePath = jest.spyOn(path, 'resolve');
+    resolvePath.mockImplementation((path: string) => path);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '8.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '8.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '8.0.289', 'x64') }
+    ],
+    [
+      { version: '11.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '11.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '11.0.289', 'x64') }
+    ],
+    [
+      { version: '12.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '12.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '12.0.289', 'x64') }
+    ],
+    [
+      { version: '13.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '13.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '13.0.289', 'x64') }
+    ],
+    [
+      { version: '15.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '15.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '15.0.289', 'x64') }
+    ],
+    [
+      { version: '16.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '16.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '16.0.289', 'x64') }
+    ],
+    [
+      { version: '7.0.289', arch: 'x64', packageType: 'jdk' },
+      expectedJdkFile,
+      { javaVersion: '7.0.289', javaPath: path.join('Java_LocalJDKFile_jdk', '7.0.289', 'x64') }
+    ],
+    [
+      { version: actualJavaVersion, arch: 'x86', packageType: 'jdk' },
+      'not_existing_one',
+      {
+        javaVersion: actualJavaVersion,
+        javaPath: path.join('Java_LocalJDKFile_jdk', actualJavaVersion, 'x86')
+      }
+    ]
+  ])('inputs %o, jdkfile %s, result %o', async (inputs, jdkFile, expected) => {
+    mockJavaBase = new LocalDistributor(inputs, jdkFile);
+    await expect(mockJavaBase.setupJava()).resolves.toEqual(expected);
+    expect(tcFind).toHaveBeenCalled();
+    if (inputs.version === actualJavaVersion) {
+      expect(tcCacheDir).not.toHaveBeenCalled();
+      expect(utilsExtractJdkFile).not.toHaveBeenCalled();
+      expect(fsStat).not.toHaveBeenCalled();
+    } else {
+      expect(tcCacheDir).toHaveBeenCalled();
+      expect(utilsExtractJdkFile).toHaveBeenCalled();
+      expect(fsStat).toHaveBeenCalled();
+    }
+  });
+
+  it.each([
+    [{ version: '8.0.289', arch: 'x64', packageType: 'jdk' }, 'otherJdkFile'],
+    [{ version: '11.0.289', arch: 'x64', packageType: 'jdk' }, 'otherJdkFile'],
+    [{ version: '12.0.289', arch: 'x64', packageType: 'jdk' }, 'otherJdkFile'],
+    [{ version: '11.1.11', arch: 'x64', packageType: 'jdk' }, 'not_existing_one']
```

Comment:
Do we really need so many cases for exception? Looks like all of them test case where file is not found. Let's leave the single one

Reply:
Done. Thank you.